internal/getproviders/types.go

```diff
@@ -422,27 +422,24 @@ func VersionConstraintsString(spec VersionConstraints) string {
 			b.WriteString("??? ")
 		}
 
+		// The parser allows writing abbreviated versions (such as ~> 2) which
+		// end up being represented in memory with trailing unconstrained parts
+		// (for example ~> 2.*).  That's not really any different than writing
+		// the constrained-to-zero version (like 2.0) and so we'll prefer that
```

Comment:
```suggestion
		// the constrained-to-zero version (like ~> 2.0) and so we'll prefer that
```

something _like_ this?